src/coreclr/src/vm/multicorejitplayer.cpp

```diff
@@ -883,6 +898,24 @@ bool MulticoreJitProfilePlayer::HandleModuleDependency(unsigned jitInfo)
                 SString assemblyName;
                 assemblyName.SetASCII(mod.m_pRecord->GetAssemblyName(), mod.m_pRecord->AssemblyNameLen());
 
+                // Handle only modules which are in TPA list
```

Comment:
Why do we want to limit it this way?

Reply:
In our environment, multicorejit tries to load modules of applications that cannot be loaded because profiles are collected in the application termination and multicorejit is performed in pre-initialization stage which is before actual application launching. (And I hope to reuse profiled data in other applications.) If modules cannot be loaded in multicorejit, I found it iscached in FailureCache and it cannot be loaded after. I think it can be removed in the PR. However, I need a similar approach to our environment. If you have a better idea, could you please share it?

Reply:
I am wondering whether we should load the assembly here without caching the failure.

Reply:
This is what I tested. I first get profile data when an application terminates. And then a process is launched. It pre-compiles methods using multicorejit before actual application launching.(It is like old zygote.) Multicorejit tries to load all modules in the profile data (It has modules of application as well as system.) However, in this stage, it doesn't know modules of application (It fails and cached.) And then when a user clicks an application, it fails to load module of application because it is cached in FailureCache even if it can be loaded.
If I misunderstand how FailureCache works, please let me know. Thank you

Reply:
Thank you! After this patch is merged, I will create a PR on it.